solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java

```diff
@@ -486,16 +486,14 @@ private Object decodeDVField(int localId, LeafReader leafReader, String fieldNam
       case SORTED_NUMERIC:
         final SortedNumericDocValues numericDv = leafReader.getSortedNumericDocValues(fieldName);
         if (numericDv != null && numericDv.advance(localId) == localId) {
-          if (schemaField.getType() instanceof LatLonPointSpatialField) {
-            long number = numericDv.nextValue();
-            return ((LatLonPointSpatialField) schemaField.getType()).geoValueToStringValue(number);
-          }
           final List<Object> outValues = new ArrayList<>(numericDv.docValueCount());
           for (int i = 0; i < numericDv.docValueCount(); i++) {
             long number = numericDv.nextValue();
             Object value = decodeNumberFromDV(schemaField, number, true);
             // return immediately if the number is not decodable, hence won't return an empty list.
             if (value == null) return null;
+            // return the value as "lat, lon" if its not multi-valued
```

Comment:
This is not consistent with how Solr normally does things.  If the field type is declared as multiValued then we normally always return as a list; otherwise we never do.  Here I think you're making that vary per document dependent on how many values the document has.

Reply:
The only reason I had to do this is LatLonDocValuesField type is SORTED_NUMERIC, so even for non-multivalued data, we would be returning an array.  If that is what is excepted?
If the field is stored then it would return string for non-multi-valued data and string array for multi-valued data.

Reply:
I understand that internally the type is SORTED_NUMERIC either way.  But externally (from user's point of view) the visible behavior should be consistent with what would happen if the field were stored.  Please ensure this distinction is tested too (if it isn't already)

Reply:
Yes, I have test case of both store and docValue